server/sonar-server/src/main/java/org/sonar/server/computation/container/ComputeEngineContainerImpl.java

```diff
@@ -154,14 +167,30 @@ private static List componentClasses() {
       NewCoverageMetricKeysModule.class,
 
       // issues
-      ScmAccountCacheLoader.class,
-      ScmAccountCache.class,
-      SourceLinesCache.class,
-      IssueComputation.class,
-      RuleCache.class,
       RuleCacheLoader.class,
+      RuleRepositoryImpl.class,
+      ScmAccountToUserLoader.class,
+      ScmAccountToUser.class,
       IssueCache.class,
+      DefaultAssignee.class,
+      IssueVisitors.class,
+      IssueLifecycle.class,
+
+      // order is important: DebtAggregator then NewDebtAggregator (new debt requires debt)
```

Comment:
if order is that much important, its unfortunate to rely on order of addition into the container to enforce it and a comment. This is very weak.
In the end, there isn't much difference between using the container with that order constraint and having an hardcoded list of IssueVisitor in IssueVisitors for example